### PR TITLE
[hotfix][FLINK-26832] Some minor logging / observe fixes

### PR DESCRIPTION
**Pull request by gyfora**

### CODE REVIEW

**Comment by gyfora** [comment]
cc @bgeng777

File: flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/observer/BaseObserver.java [hunk 1]
```diff
@@ -103,10 +111,12 @@ protected void observeJmDeployment(
                 return;
             }
 
+            logger.info("JobManager is deployed but not ready yet", status);
```
Comment (bgeng777) [inline flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/observer/BaseObserver.java:114]: The old code seems more clear:
```
logger.info("JobManager deployment exists but not ready, status {}", status);
```
The new description `deployed but not ready yet` is not wrong but is too close to the enum `DEPLOYED_NOT_READY`, while we actully set the status to `DEPLOYING`.

Comment (gyfora) [reply flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/observer/BaseObserver.java:114]: At this point it is always deploying status so no point in printing the current status. I could change it to 
```
JobManager is being deployed
```

Comment (gyfora) [reply flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/observer/BaseObserver.java]: fixed

**Comment by bgeng777** [comment]
Thanks for improving the logging! LGTM.

The optimization of skipping observation of JM when the job is in `suspended` state can reduce unnecessary checks, as we will not care JM when there is no job running. 
I am not perfectly sure if we need to make some changes for incoming session job. But I think it is out of the scope of this PR.

**Pull request merged**